vms/rpcchainvm/vm.go

```diff
@@ -46,6 +49,11 @@ func Serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption
 		for {
 			select {
 			case s := <-signals:
+				if !allowShutdown.Get() {
+					fmt.Printf("runtime engine: ignoring signal: %s\n", s)
+					continue
+				}
+
 				switch s {
 				case syscall.SIGINT:
 					fmt.Println("runtime engine: ignoring signal: SIGINT")
```

Comment:
super minor but the string representation of the signal is `terminate` for `SIGTERM` above 
```suggestion
					fmt.Println("runtime engine: ignoring signal: interrupt")
```